README.md

```diff
@@ -1 +1,249 @@
-# openapi-to-fastapi
\ No newline at end of file
+## The reasoning
+
+There's an awesome [FastAPI](https://github.com/tiangolo/fastapi) library which simplifies the process of creating APIs.
+The one of the most exciting parts of it is the generation of OpenAPI specs out of the box.
+But what if.. you have an OpenAPI spec and you need to create an API from it?
+
+One day we met such problem — the need of creating an API from a multiple OpenAPI specs,
+and to make sure that requests which comes in and the responses which comes out are aligned
+with the models defined in those specs.
+
+> ⚠️ This library was created with an idea to cover only our needs first.
+> So for now it's not suitable for everyone and has a lot of technical restrictions.
+> Please consider it as an experimental stuff
+
+## Installation
+
+The package is available on PyPi:
+
+```bash
+pip install openapi-to-fastapi
+```
+
+## FastAPI routes generation
+
+The main purpose of this library is to generate FastAPI routes from the OpenAPI specs.
+This is done by:
+
+```python
+from pathlib import Path
+from openapi_to_fastapi.routes import make_router_from_specs
+
+specs = Path("./specs")
+
+router = make_router_from_specs(specs)
+```
+
+The code above will generate FastAPI router which then can be included into the main router,
+or just be used as the default one.
+
+Imagine you have a following spec (some parts are cut off for brevity):
+
+```json
+{
+  "openapi": "3.0.2",
+   ...
+  "paths": {
+    "/Company/BasicInfo": {
+      "post": {
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/BasicCompanyInfoRequest"
+    ...
+        "responses": {
+          "200": {
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/BasicCompanyInfoResponse"
+
+    ...
+  },
+  "components": {
+    "schemas": {
+      "BasicCompanyInfoRequest": {
+        "title": "BasicCompanyInfoRequest",
+        "required": ["companyId"],
+        "type": "object",
+        "properties": {
+          "companyId": {
+            "title": "Company Id",
+            "type": "string",
+            "example": "2464491-9"
+          }
+        }
+      },
+      "BasicCompanyInfoResponse": {
+        "title": "BasicCompanyInfoResponse",
+        "required": ["name", "companyId", "companyForm"],
+        "type": "object",
+        "properties": {
+          "name": {
+            "title": "Name of the company",
+            "type": "string",
+          },
+          "companyId": {
+            "title": "ID of the company",
+            "type": "string",
+          },
+          "companyForm": {
+            "title": "The company form of the company",
+            "type": "string",
+          },
+```
```

Comment:
it would probably be a good idea to still do the closing `}` etc. when you do cut things off for brevity because as it is it will confuse a lot of tools